### ISSUE TITLE
Updating package with VS GUI removes "type" attribute

### ISSUE THREAD

**Issue by natemcmaster** [NuGet/Home#3690]
## Details about Problem

NuGet product used:  VS UI
NuGet version: 3.5.0.1484
dotnet.exe --version: 1.0.0-preview2-1-003153
VS version: enterprise 2015, update3
## Repro
1. Use xproj/project.json with dependencies that aren't the latest, and that have "type": "anything" defined.
   
   ``` json
   {
     "dependencies": {
       "Microsoft.NETCore.App": {
         "type": "platform",
         "version": "1.0.0"
       },
       "Microsoft.Extensions.Logging": {
         "type": "build",
         "version": "1.0.0"
       }
     },
     "frameworks": {
       "netcoreapp1.0": {}
     }
   }
   ```
2. Upgrade package with VS GUI

**Result**
Package upgrade to new version, but it removed the "type" attribute.

``` json
{
   "dependencies": {
       "Microsoft.NETCore.App": "1.1.0-preview1-001100-00",
       "Microsoft.Extensions.Logging": "1.1.0-preview1-final"
    },
    "frameworks": {
        "netcoreapp1.0": {}
    }
}
```

**Expected**
Upgrade version and leave "type" attribute intact.

**Comment by rrelyea** [comment]
@joelverhagen  - didn't we address this? or is this slightly different?

**Comment by joelverhagen** [comment]
The fix is not in `3.5.0-beta2` (build 1484). It is in `3.5.0-rc1`. @natemcmaster, retry this on `3.5.0-rc1` from:
http://dist.nuget.org/index.html

**Comment by natemcmaster** [comment]
I tried with 3.5.0-rc1 (3.5.0.173) and it's still happening.

**Comment by joelverhagen** [comment]
Thanks for checking @natemcmaster, I'll investigate.

**Comment by joelverhagen** [comment]
@natemcmaster, the full fix is in `3.5.0-rtm` (not on dist yet). There was an initial change that fixed a very specific scenario and a follow-up change that covered all updates in VS (all tabs, `Update-Package` in PMC, etc). @rrelyea can speak to release dates and such.

**Comment by natemcmaster** [comment]
I expect this is going to cause some confusing when the 1.1 .NET Core upgrade ships. A .NET Core user upgrading a .NET Core app from 1.0 to 1.1 with the nuget GUI is likely to get this error:

```
Can not find runtime target for framework '.NETCoreApp,Version=v1.1' compatible with one of the target runtimes: 'win10-x64, win81-x64, win8-x64, win7-x64'. Possible causes:
1. The project has not been restored or restore failed - run `dotnet restore`
2. The project does not list one of 'win10-x64, win81-x64, win8-x64, win7-x64' in the 'runtimes' section.
3. You may be trying to publish a library, which is not supported. Use `dotnet pack` to distribute libraries.
```

The error doesn't make it clear that the resolution is to put `"type": "platform"` back in.

cc @danroth27 @piotrpMSFT

**Comment by rrelyea** [comment]
My guess is that Net Core tools that ship with 1.1 should update to chain install 3.5.0-rtm. Will create internal mail thread.

**Comment by danroth27** [comment]
@leecow

**Comment by joelverhagen** [comment]
This is fixed in 3.5.0-rtm and later.

**Comment by joelverhagen** [comment]
Duplicate of https://github.com/NuGet/Home/issues/3303.
